todoapp/src/components/Todos/Todos.vue

```diff
@@ -0,0 +1,124 @@
+      <li v-for="(todo, index) in sortedAndFilteredTodos" :key="todo.id">
+        <Todo
+          :todo="todo"
+          :index="index"
+          :current-edited-todo="currentEditedTodo"
+          @get-todo-id="getTodoId"
+          @on-todo-update="updateTodo"
+          @display-popup="displayPopup"
+          @mark-todo-status="markTodoStatus"
+        />
+      </li>
+    </ul>
+    <NoTodosFound
+      v-else-if="todos.length"
+      :search-input-content="searchInput"
+      @add-search-todo="addSearchTodo"
+    />
+    <EmptyState v-if="!todos.length" :todos="todos" />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue'
+import Todo from './Todo.vue'
+import EmptyState from '../EmptyState/EmptyState.vue'
+import type { TodoType } from '../../types/text'
+import type { SelectedState } from '../../types/selected'
+import NoTodosFound from '../EmptyState/NoTodosFound.vue'
+
+const props = defineProps<{
+  todos: TodoType[]
+  searchInput: string
+  searchTodos: TodoType[]
+  selectedSortingButtons: SelectedState
+  currentEditedTodo: number | null
+}>()
+
+const emit = defineEmits<{
+  (e: 'getTodoId', id: number, isFinishedTodo: boolean): void
+  (e: 'getId', id: number, isFinishedTodo: boolean): void
+  (e: 'displayPopup', isFinishedTodo: boolean): void
+  (e: 'markTodoStatus', todo: TodoType): void
+  (e: 'onTodoUpdate', todo: TodoType): void
+  (e: 'clearSearchInput'): void
+  (e: 'addSearchTodo'): void
+}>()
+
+function updateTodo(todo: TodoType) {
+  emit('onTodoUpdate', todo)
+}
+
+function getTodoId(id: number, isFinishedTodo: boolean) {
+  emit('getTodoId', id, isFinishedTodo)
+}
+
+function markTodoStatus(todo: TodoType) {
+  emit('markTodoStatus', todo)
+}
+
+function displayPopup(isFinishedTodo: boolean) {
+  emit('displayPopup', isFinishedTodo)
+}
+
+function addSearchTodo() {
+  emit('addSearchTodo')
+}
+
+function trimWhiteSpace(str: string) {
+  return str.trim()
+}
+
+const sortedAndFilteredTodos = computed(() => {
+  const selectedButton = Object.keys(props.selectedSortingButtons).find(
+    (key) => props.selectedSortingButtons[key as keyof SelectedState].selected
+  )
+
+  if (selectedButton === 'title') {
+    return props.searchTodos
+      .slice()
+      .sort((a, b) =>
+        props.selectedSortingButtons.title.order
+          ? trimWhiteSpace(a.title).localeCompare(trimWhiteSpace(b.title))
+          : trimWhiteSpace(b.title).localeCompare(trimWhiteSpace(a.title))
+      )
+  } else if (selectedButton === 'description') {
+    return props.searchTodos
+      .slice()
+      .sort((a, b) =>
+        props.selectedSortingButtons.description.order
+          ? trimWhiteSpace(a.text).localeCompare(trimWhiteSpace(b.text))
+          : trimWhiteSpace(b.text).localeCompare(trimWhiteSpace(a.text))
+      )
+  } else if (selectedButton === 'date') {
+    return props.searchTodos.slice().sort((a, b) => {
+      const dateA = new Date(
```

Comment:
Please review this if..else statement and try to create general local functions for the code duplications.

Reply:
The `Todos.vue` and `FinishedTodos.vue` are almost the same. You should create a general component for the common blocks and use it in the `Todos.vue` and `FinishedTodos.vue` instead of duplicating the logic.